ui/src/shared/components/CloudUpgradeButton.tsx

```diff
@@ -1,25 +1,56 @@
 // Libraries
 import React, {FC} from 'react'
 import {Link} from 'react-router'
+import {connect} from 'react-redux'
 
 // Components
 import CloudOnly from 'src/shared/components/cloud/CloudOnly'
 
 // Constants
 import {CLOUD_URL, CLOUD_CHECKOUT_PATH} from 'src/shared/constants'
+import {
+  HIDE_UPGRADE_CTA_KEY,
+  PAID_ORG_HIDE_UPGRADE_SETTING,
+} from 'src/cloud/constants'
 
-const CloudUpgradeButton: FC = () => {
+// Types
+import {AppState, OrgSetting} from 'src/types'
+
+interface StateProps {
+  show: boolean
+}
+
+const CloudUpgradeButton: FC<StateProps> = ({show}) => {
   return (
     <CloudOnly>
-      <Link
-        className="cf-button cf-button-sm cf-button-success upgrade-payg--button"
-        to={`${CLOUD_URL}${CLOUD_CHECKOUT_PATH}`}
-        target="_self"
-      >
-        Upgrade Now
-      </Link>
+      {show ? (
```

Comment:
seems like it's _already_ taking care of it's own display state. if you move the OrgSettings logic in here with a useEffect to launch the action on mounting, you can remove all of the code changes in the TreeNav component

Reply:
The OrgSettings component is only responsible for the action that makes the call for the org settings from IDPE. Ideally, this should be done just once.

There are two separate components, CloudUpgradeButton and CloudUpgradeNavBanner, that prompt the user to "Upgrade Now". Each of these components should **not** be making their own identical calls to IDPE for the same resource. They should just receive props from state for that resource.

That's why the logic is separated. <OrgSettings> is really just a wrapper to fetch the resource. Fetching the resource just once, and having a component (TreeNav) that is reasonably expected to appear only once on the page seemed like an ideal marriage.

And then all of the child components on the page are supported with the resource.